src/java.base/share/classes/sun/net/www/MimeTable.java

```diff
@@ -155,22 +160,41 @@ public synchronized MimeEntry find(String type) {
      * with it. Parses general file names, and URLs.
      */
     public MimeEntry findByFileName(String fname) {
-        String ext = "";
+        // attempt to find the entry with the fragment component removed
+        MimeEntry entry = findByFileName(fname, true);
 
-        int i = fname.lastIndexOf('#');
+        // if entry not found, try again with the fragment intact
+        if (entry == null)
+            entry = findByFileName(fname, false);
 
-        if (i > 0) {
-            fname = fname.substring(0, i - 1);
+        return entry;
+    }
+
+    /**
+     * Locate a MimeEntry by its associated file extension.
+     *
+     * @param fname the file name
+     *
+     * @param removeFragment whether to remove the fragment, if any,
+     *        comprising the last hash ('#') and any subsequent characters
+     *
+     * @return the MIME entry associated with the file name
+     */
+    public MimeEntry findByFileName(String fname, boolean removeFragment) {
```

Comment:
Hello Brian,

Perhaps this new method can be made `private` (and maybe even `static`)?

Reply:
Private yes, static no. Refactored in eab33e8000ea730c57c918dbf291af23bacbd059.